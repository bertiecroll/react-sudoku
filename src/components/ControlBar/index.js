export { ControlBar } from './ControlBar'
export { default } from './ControlBar'
