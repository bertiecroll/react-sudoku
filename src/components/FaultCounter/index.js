export { FaultCounter } from './FaultCounter'
export { default } from './FaultCounter'
