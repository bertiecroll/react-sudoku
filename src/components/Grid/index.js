export { Grid } from './Grid'
export { GridRow } from './GridRow'
