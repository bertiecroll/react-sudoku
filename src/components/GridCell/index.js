export { GridCell } from './GridCell'
export { EmptyGridCell } from './EmptyGridCell'
export { CompleteGridCell } from './CompleteGridCell'
export { default } from './GridCell'
