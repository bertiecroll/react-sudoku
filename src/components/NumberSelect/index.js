export { NumberSelect } from './NumberSelect'
export { default } from './NumberSelect'
