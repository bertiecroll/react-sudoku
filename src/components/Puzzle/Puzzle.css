.Puzzle {
  margin: 1em;
  padding: 1em;
  display: flex;
  flex-direction: column;
  align-items: center;
}
