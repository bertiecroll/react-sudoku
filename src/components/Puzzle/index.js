export { Puzzle } from './Puzzle'
export { default } from './Puzzle'
