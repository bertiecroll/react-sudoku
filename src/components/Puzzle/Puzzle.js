import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import './Puzzle.css'

import { Grid } from 'components/Grid'
import { cellsByRow } from 'redux/selectors/cells'
import ControlBar from 'components/ControlBar'
import ProgressBar from 'components/ProgressBar'
import Spinner from 'components/Spinner'

import { generateCells } from 'redux/actions/cells'
import { isFetching } from 'redux/selectors/cells'

export class Puzzle extends Component {

  componentDidMount () {
    const difficultyLevel = this.props.match.params.difficultyLevel
    difficultyLevel && this.props.generateCells(difficultyLevel.toUpperCase())
  }

  render() {
    return this.props.isLoading
    ? (
      <div className="Puzzle">
        <Spinner />
      </div>
    )
    : (
      <div className="Puzzle">
        <ProgressBar />
        <Grid cellsByRow={this.props.cellsByRow} />
        <ControlBar />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  cellsByRow: cellsByRow(state),
  isLoading: isFetching(state),
})

const mapDispatchToProps = { generateCells }

export default connect(mapStateToProps, mapDispatchToProps)(Puzzle)

Puzzle.defaultProps = {
  cellsByRow: {}
}

Puzzle.propTypes = {
  cellsByRow: PropTypes.object.isRequired,
}
