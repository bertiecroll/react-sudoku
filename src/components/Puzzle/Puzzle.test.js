import React from 'react'
import { shallow } from 'enzyme'
import { Puzzle } from 'components/Puzzle'

const routerMatch = { params: { difficultyLevel: 'test ' } }
const mockGenerateCells = jest.fn()

describe('Puzzle', () => {
  it('renders Spinner component when props.puzzleStatus is LOADING', () => {
    const wrapper = shallow(
      <Puzzle puzzleStatus='LOADING' match={routerMatch} generateCells={mockGenerateCells} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('renders ProgressBar, Grid and Control Bar when props.puzzleStatus is IN_PROGRESS', () => {
    const wrapper = shallow(
        <Puzzle puzzleStatus='IN_PROGRESS' match={routerMatch} generateCells={mockGenerateCells} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders Result when props.puzzleStatus is COMPLETED', () => {
    const wrapper = shallow(
      <Puzzle puzzleStatus='COMPLETED' match={routerMatch} generateCells={mockGenerateCells} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
