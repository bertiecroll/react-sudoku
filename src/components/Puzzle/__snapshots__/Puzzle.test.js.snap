// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puzzle renders ProgressBar, Grid and Control Bar when props.puzzleStatus is IN_PROGRESS 1`] = `
<div
  className="Puzzle"
>
  <ProgressBar />
  <Grid
    cellsByRow={Object {}}
  />
  <Connect(ControlBar) />
</div>
`;

exports[`Puzzle renders Result when props.puzzleStatus is COMPLETED 1`] = `
<div
  className="Puzzle"
>
  <Result />
</div>
`;

exports[`Puzzle renders Spinner component when props.puzzleStatus is LOADING 1`] = `
<div
  className="Puzzle"
>
  <Spinner />
</div>
`;
